src/builder.ts

```diff
@@ -869,6 +931,7 @@ export class SchemaBuilder {
         ...config.extensions,
         nexus: new NexusInputObjectTypeExtension(config),
       },
+      ...this.maybeAddDirectiveUses('INPUT_OBJECT', config.directives),
```

Comment:
I said it in the `makeSchema`, but I want to drive the point, since this adds a whole `astNode` I feel that someone that might want to modify the AST node for another purpose will get blocked by this code or might override it entirely. I think we should at the very least wrap that in an array and a merge function or even better if only the `directives` field of the node was returned by `maybeAddDirectiveUses` (free and member functions) and the `astNode` was generated by another method.

Reply:
> I feel that someone that might want to modify the AST node for another purpose will get blocked by this code or might override it entirely

Yeah, fair points - I had considered that and was thinking of just crossing that bridge if/when it came to it, since `astNode` isn't part of the public api for any of the nexus builder fn's, I wasn't sure if that would actually happen in practice.

Reply:
In theory no, we should never touch the AST node since it is supposed to the result of the parsing of the schema.
I still think it should be explicit in the code that we are converting the directives to an AST node. Something like:
```typescript
this.createAstNode([this.maybeAddDirectiveUses])
```
I suspect that a future release of graphql will add backend directives in the spec, so it will just be easier to move to that then.

Reply:
@tgriesser what would be the shortest path to unblock this pull request so it can go through?

Reply:
@zingerj this is the blocker.

Reply:
Going to punt on this for now - going to assume that if you need to modify the ASTNode you can use something else more specialized like graphql-tools